### PR TITLE
Add a mini keyboard: Dosa40RGB

### PR DESCRIPTION
**Pull request by daotakisaac**
## Description

Add a 40 percent keyboard: Dosa40RGB BLE

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dtisaac/dosa40rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_DI_PIN D0
+#define DRIVER_LED_TOTAL 42
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED false // turn off effects when suspended
```
Comment (drashna) [inline keyboards/dtisaac/dosa40rgb/config.h:62]:
```suggestion
// #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/dtisaac/dosa40rgb/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH = AdafruitBLE     # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/dtisaac/dosa40rgb/rules.mk:24]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
BLUETOOTH = AdafruitBLE     # Enable Bluetooth
```

File: keyboards/dtisaac/dosa40rgb/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+* Keyboard Maintainer: [DTIsaac](https://github.com/daotakisaac)
+* Hardware Supported: ATmega32U4 + AdafruitBLE SPI
+
+Make example for this keyboard (after setting up your build environment):
+
+    make dosa40rgb:default
+
+Flashing example for this keyboard:
+
+    make dosa40rgb:default:flash
```
Comment (fauxpark) [inline keyboards/dtisaac/dosa40rgb/readme.md:16]:
```suggestion
    make dtisaac/dosa40rgb:default

Flashing example for this keyboard:

    make dtisaac/dosa40rgb:default:flash
```

File: keyboards/dtisaac/dosa40rgb/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                 LED_EN,          RGB_RMOD, KC_UP,     RGB_MOD,   RGB_HUI,  RGB_VAI,  RGB_SAI,  RGB_SPI,  KC_TRNS,  OUT_USB,  OUT_BT,  BLE_DIS,
+                 KC_TRNS,         KC_LEFT,  KC_DOWN,   KC_RGHT,   RGB_HUD,  RGB_VAD,  RGB_SAD,  RGB_SPD,  KC_TRNS,  KC_TRNS,  KC_BSLS,
+                 KC_TRNS,         KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+                 KC_TRNS,         KC_TRNS,  KC_TRNS,   			  KC_TAB,             KC_TRNS,  		  KC_TRNS,  KC_TRNS,  KC_TRNS
+    ),
+};
+
+void rgb_matrix_indicators_user(void)
+{
+ 	if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK))
```
Comment (fauxpark) [inline keyboards/dtisaac/dosa40rgb/keymaps/default/keymap.c:64]:
```suggestion
 	if (host_keyboard_led_state().caps_lock)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by daotakisaac** [comment]
Please help me check it again!!

File: keyboards/dtisaac/dosa40rgb/info.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{
+    "keyboard_name": ">_Dosa40rgb",
+    "url": "",
+    "maintainer": "DTIsaac",
+    "width": 12,
+    "height": 4,
```
Comment (fauxpark) [inline keyboards/dtisaac/dosa40rgb/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by daotakisaac** [comment]
Please help me!!!

**Comment by daotakisaac** [comment]
Bump again for review, thanks!

**Comment by daotakisaac** [comment]
Does the file have any errors?
Please help me check it, thanks!

**Comment by daotakisaac** [comment]
**### Hello, please help me!!!  I waited so long.......**

**Comment by drashna** [comment]
This has a number of merge conflicts that need to be resolved.

**Comment by daotakisaac** [comment]
Do not need to merge anymore

**Pull request closed without merging**